File: content/problems/2609-find-the-longest-balanced-substring-of-a-binary-string/solutions/scan-zero-one-runs/Solution.kt
class Solution {
    fun findTheLongestBalancedSubstring(s:String):Int{
        var index=0;
        var answer=0;
        while(index<s.length){
            var zeros=0;
            var ones=0;
            while(index<s.length&&s[index]=='0'){
                zeros++;
                index++
            };
            while(index<s.length&&s[index]=='1'){
                ones++;
                index++
            };
            answer=maxOf(answer,2*minOf(zeros,ones))
        };
        return answer
    }
}

File: content/problems/2609-find-the-longest-balanced-substring-of-a-binary-string/solutions/scan-zero-one-runs/solution.cpp
class Solution {
    public:int findTheLongestBalancedSubstring(string s){
        int index=0,answer=0;
        while(index<(int)s.size()){
            int zeros=0,ones=0;
            while(index<(int)s.size()&&s[index]=='0'){
                ++zeros;
                ++index;
            }while(index<(int)s.size()&&s[index]=='1'){
                ++ones;
                ++index;
            }
            answer=max(answer,2*min(zeros,ones));
        }
        return answer;
    }
};
